### PR TITLE
Issue 474 - Mapping currency -> locale is not unique

### PR DESCRIPTION
**Pull request by javier-gomez-github**
- Changed code in DefaultInvoiceFormatter to match the new Currencies and Symbols defined in killbill-api
- Fixed Integration Tests to match the new symbols from killbill-api

### CODE REVIEW

File: invoice/src/main/java/org/killbill/billing/invoice/template/formatters/DefaultInvoiceFormatter.java [hunk 1]
```diff
@@ -240,8 +240,8 @@ private String getFormattedAmountByLocaleAndInvoiceCurrency(final BigDecimal amo
         dfs.setInternationalCurrencySymbol(currencyUnit.getCurrencyCode());
 
         try {
-            final java.util.Currency currency = java.util.Currency.getInstance(invoiceCurrencyCode);
-            dfs.setCurrencySymbol(currency.getSymbol(currencyLocaleMap.get(currency)));
```
Comment (pierre) [inline invoice/src/main/java/org/killbill/billing/invoice/template/formatters/DefaultInvoiceFormatter.java]: We should now be able to get rid of the `currencyLocaleMap`.

Comment (javier-gomez-github) [inline invoice/src/main/java/org/killbill/billing/invoice/template/formatters/DefaultInvoiceFormatter.java]: Done!

**Comment by sbrossie** [comment]
@javier-gomez-github After merging i am seeing a compilation error in `DefaultInvoiceFormatterFactory`: Should we get rid of the map [here](https://github.com/killbill/killbill/blob/work-for-release-0.17.x/invoice/src/main/java/org/killbill/billing/invoice/template/formatters/DefaultInvoiceFormatterFactory.java#L52) as well?

**Comment by pierre** [comment]
> Should we get rid of the map [here](https://github.com/killbill/killbill/blob/work-for-release-0.17.x/invoice/src/main/java/org/killbill/billing/invoice/template/formatters/DefaultInvoiceFormatterFactory.java#L52) as well?

Yup, that map was a hack that this issue fixes.

**Pull request merged**